### PR TITLE
remove @show and make print conditional

### PR DESCRIPTION
**Pull request by szy21**
# PULL REQUEST

## Purpose and Content
    (1) Remove @show statements
    (2) Update solver function to avoid entering iterative loop in neutral
    conditions
    (3) Update exchange coefficients to avoid iterative loop in neutral
    conditions
    (4) Update convergence coverage checks to maintain physically reasonable ratio of z0m/z0b
    (5) Return FT(Inf) in neutral conditions
            modified:   src/SurfaceFluxes.jl
            modified:   test/runtests.jl
            modified:   test/test_convergence.jl
    + Format

Refactors test (add F64 checks), removes unnecessary function imports from Thermodynamics->TestedProfiles, clears code duplication,re-enable GPU checks, increase coverage of convergence checks. 

## Benefits and Risks
Risks: Package requires further physics checks and tests in breaking edge cases ("data-tests" similar to Thermodynamics.jl will be added when such cases are discovered in integration tests with TC.jl and ClimaAtmos.jl)

### CODE REVIEW

File: test/test_convergence.jl [hunk 1]
```diff
@@ -23,77 +23,9 @@ const CP = CLIMAParameters
...
-        RH,
-        e_pot,
-        u,
-        v,
-        w,
-        e_kin,
-        phase_type,
-    )
-end
 function input_config(ArrayType; n = 10, n_RS1 = 20, n_RS2 = 20, T_surface = 290, T_min = 150)
```
Comment (akshaysridhar) [inline test/test_convergence.jl:29]: @charleskawczynski The purpose of this `input_config` here is to overload the default temperatures in the `TestedProfiles` definition ; can we pass an array of temperatures here?

File: test/test_convergence.jl [hunk 2]
```diff
@@ -103,111 +35,76 @@ function input_config(ArrayType; n = 10, n_RS1 = 20, n_RS2 = 20, T_surface = 290
...
-    else
-        push!(profiles_int, prof)
-    end
+function generate_profiles(FT)
+    toml_dict = CP.create_toml_dict(FT; dict_type = "alias")
+    param_set = create_parameters(toml_dict, UF.BusingerType())
+    thermo_params = SFP.thermodynamics_params(param_set)
+    uft = SFP.universal_func_type(param_set)
+    profiles = collect(TD.TestedProfiles.PhaseEquilProfiles(thermo_params, Array{FT}))
+    profiles_sfc = filter(p->iszero(p.z), profiles)
```
Comment (akshaysridhar) [inline test/test_convergence.jl:44]: @charleskawczynski `push`ing to generic container swapped out for `filter`ed collection.

File: src/SurfaceFluxes.jl [hunk 3]
```diff
@@ -383,56 +378,54 @@ function obukhov_length(
...
+            KA.@print(", q_sfc = ", TD.total_specific_humidity(thermo_params, ts_sfc(sc)))
+            KA.@print(", u_in = ", u_in(sc))
+            KA.@print(", u_sfc = ", u_sfc(sc))
+            KA.@print(", z0_m = ", z0(sc, UF.MomentumTransport()))
+            KA.@print(", z0_b = ", z0(sc, UF.HeatTransport()))
+            KA.@print(", Δz = ", Δz(sc))
+            KA.@print(", ΔDSE = ", ΔDSE)
             if soltype isa CompactSolution
-                @show sol.root
+                KA.@print(", sol.root = ", sol.root)
```
Comment (akshaysridhar) [inline src/SurfaceFluxes.jl:406]: @charleskawczynski `KA.@print` replaces the `@show` statements here

File: src/SurfaceFluxes.jl [hunk 4]
```diff
@@ -95,7 +88,6 @@ function Base.show(io::IO, sfc::SurfaceFluxConditions)
     println(io, "C_drag                 = ", sfc.Cd)
     println(io, "C_heat                 = ", sfc.Ch)
     println(io, "evaporation            = ", sfc.evaporation)
```
Comment (akshaysridhar) [inline src/SurfaceFluxes.jl:90]: @charleskawczynski `is_converged` flag removed ; now assumes that `error_on_non_convergence() == true`, such that the simulation is interrupted if the convergence criteria aren't met.

File: test/test_convergence.jl (hunk 1 above)
Comment (szy21) [reply test/test_convergence.jl:29]: It seems `input_config` is not used in this file?

Comment (charleskawczynski) [reply test/test_convergence.jl:29]: Was this just needed for local experimentation? If so, we can open a PR in Thermodynamics that forwards `input_config_kwargs` from `PhaseEquilProfiles` to `input_config`.

File: src/SurfaceFluxes.jl [hunk 5]
```diff
@@ -383,56 +378,54 @@ function obukhov_length(
...
+            KA.@print(", T_in = ", TD.air_temperature(thermo_params, ts_in(sc)))
+            KA.@print(", T_sfc = ", TD.air_temperature(thermo_params, ts_sfc(sc)))
+            KA.@print(", q_in = ", TD.total_specific_humidity(thermo_params, ts_in(sc)))
+            KA.@print(", q_sfc = ", TD.total_specific_humidity(thermo_params, ts_sfc(sc)))
+            KA.@print(", u_in = ", u_in(sc))
+            KA.@print(", u_sfc = ", u_sfc(sc))
+            KA.@print(", z0_m = ", z0(sc, UF.MomentumTransport()))
+            KA.@print(", z0_b = ", z0(sc, UF.HeatTransport()))
+            KA.@print(", Δz = ", Δz(sc))
+            KA.@print(", ΔDSE = ", ΔDSE)
```
Comment (szy21) [inline src/SurfaceFluxes.jl:404]:
```suggestion
            KA.@print(", ΔDSEᵥ = ", ΔDSEᵥ)
```

File: test/test_convergence.jl (hunk 1 above)
Comment (akshaysridhar) [reply test/test_convergence.jl]: For now we are fine using the `input_configs` overload here; we can discuss the changes to Thermodynamics in the next several days.

**Comment by akshaysridhar** [comment]
bors try

**Review by akshaysridhar — APPROVED** [review]

File: src/SurfaceFluxes.jl [hunk 6]
```diff
@@ -365,74 +360,66 @@ function obukhov_length(
...
-            sol = nothing
-            is_converged = true
-        else
-            sol = RS.find_zero(root_l_mo, RS.NewtonsMethodAD(sc.L_MO_init), soltype, tol, maxiter)
-            L_MO = sol.root
-            is_converged = sol.converged
+    if abs(ΔDSEᵥ) <= tol_neutral
+        # Large L_MO -> virtual dry static energy suggests neutral boundary layer
+        # Return ζ->0 in the neutral boundary layer case, where ζ = z / L_MO
+        return L_MO = FT(Inf * sign(ΔDSEᵥ))
```
Comment (charleskawczynski) [inline src/SurfaceFluxes.jl:375]: `L_MO` is undefined, here, so this branch is broken

Comment (charleskawczynski) [reply src/SurfaceFluxes.jl:375]: Ah, nevermind, this is an assignment

File: src/SurfaceFluxes.jl [hunk 7]
```diff
@@ -365,74 +360,66 @@ function obukhov_length(
...
+                else
+                    KA.@print(", sol.root_history = ", sol.root_history)
+                    KA.@print(", sol.err_history = ", sol.err_history)
+                end
+                error("Unconverged Surface Fluxes.")
             else
-                @show sol.err_history
-                @show sol.root_history
-                KA.@print("-----------------------------------------\n")
+                KA.@print("Warning: Unconverged Surface Fluxes. Returning last interation.")
```
Comment (charleskawczynski) [inline src/SurfaceFluxes.jl:406]: We should probably remove this. What I've seen happen when doing this is that jobs appear to hang due to printing enormous logs

Comment (szy21) [reply src/SurfaceFluxes.jl:406]: So what you suggest is that we either break the simulation and print the information, or continue the simulation without printing anything?

Comment (charleskawczynski) [reply src/SurfaceFluxes.jl:406]: Yes, I suggest printing the error info and erroring out. We probably shouldn't ever run with `error_on_non_convergence() = false`. I believe that this was added to mirror thermodynamics, which is useful for writing JET tests. We'll likely need this here, too.

Comment (szy21) [reply src/SurfaceFluxes.jl:406]: Ok, It just would be nice to have an option to not erroring out when it doesn't converge for now. But we can remove it.
